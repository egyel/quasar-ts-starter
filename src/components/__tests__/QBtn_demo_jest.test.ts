import { createLocalVue, shallowMount } from '@vue/test-utils';
import { Quasar, QBtn } from 'quasar';

import BtnTest from '../QBtn_demo.vue';

const localVue = createLocalVue();
localVue.use(Quasar, { components: { QBtn } });

// used `shallowMount` instead of `mount`, will stub all **registered** components into the template
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const factory = (propsData: any = {}) => shallowMount(BtnTest, {
  localVue,
  propsData,
});

describe('BtnTest', () => {
  const wrapper = factory();
  const { vm } = wrapper;

  // test('is a Vue instance', () => {
  //   const wrapper = factory(); // <= when no props are needed
  //   // const wrapper = factory({ propName: propValue }); <= when props are needed
  //   expect(wrapper.isVueInstance()).toBeTruthy();
  // });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // or
  test('accesses the shallowMount', () => {
    expect(vm.$el.textContent).toContain('rocket muffin');
    expect(wrapper.text()).toContain('rocket muffin'); // easier
    expect(wrapper.find('p').text()).toContain('rocket muffin');
  });

  it('sets the correct default data', () => {
    const defaultData = vm.$data;
    expect(defaultData).toStrictEqual({ counter: 0, input: 'rocket muffin' });
  });
});
