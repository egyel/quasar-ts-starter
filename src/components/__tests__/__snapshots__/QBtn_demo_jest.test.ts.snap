// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnTest renders correctly 1`] = `
<div>
  <p
    class="textContent"
  >
    rocket muffin
  </p>
   
  <span>
    0
  </span>
   
  <q-btn-stub
    align="center"
    color="primary"
    icon="check"
    id="mybutton"
    label="OK"
    ripple="true"
  />
</div>
`;
